application/models/post.go

```diff
@@ -811,28 +810,37 @@ func (p *Posts) FindByUser(ctx context.Context, user User) error {
 			SELECT id FROM organizations WHERE id IN (
 				SELECT secondary_id FROM organization_trusts WHERE primary_id IN (SELECT id FROM o)
 			)
-		) AND visibility IN (?, ?)
+		) AND visibility = ?
 	)
-	AND status not in (?, ?) ORDER BY created_at desc`,
-		user.ID, PostVisibilityAll, PostVisibilityAll, PostVisibilityTrusted, PostStatusRemoved, PostStatusCompleted)
-	if err := q.All(p); err != nil {
+	AND status not in (?, ?)`
+
+	args := []interface{}{user.ID, PostVisibilityAll, PostVisibilityTrusted, PostStatusRemoved,
+		PostStatusCompleted}
+
+	if searchText != nil {
+		selectClause = selectClause + " AND (LOWER(title) LIKE ? or LOWER(description) LIKE ?)"
+		likeText := "%" + strings.ToLower(*searchText) + "%"
+		args = append(args, likeText)
+		args = append(args, likeText)
+	}
+
+	posts := Posts{}
+	q := DB.RawQuery(selectClause+" ORDER BY created_at desc", args...)
+	if err := q.All(&posts); err != nil {
 		return fmt.Errorf("error finding posts for user %s, %s", user.UUID.String(), err)
 	}
-	return nil
-}
 
-// FilterByUserTypeAndContents finds all posts belonging to the same organization as the given user,
-// not marked as completed or removed and containing a certain search text.
-func (p *Posts) FilterByUserTypeAndContents(ctx context.Context, user User, pType PostType, contains string) error {
-	where := "type = ? and (LOWER(title) like ? or LOWER(description) like ?)"
-	contains = `%` + strings.ToLower(contains) + `%`
+	if destination != nil {
+		posts = posts.FilterDestination(*destination)
+	}
+	if origin != nil {
+		posts = posts.FilterOrigin(*origin)
+	}
 
-	return DB.
-		Scope(scopeUserOrgs(user)).
-		Scope(scopeNotCompleted()).
-		Where(where, pType, contains, contains).
-		Order("created_at desc").
-		All(p)
+	for i := range posts {
+		*p = append(*p, posts[i])
```

Comment:
It looks like this assumes that p begins as an empty slice. Should we error if it's not?

Reply:
Probably, yeah.

Reply:
Or what if I just empty it out instead of error it?

Reply:
That sounds good.